internal/core/metadata/v2/application/notify.go

```diff
@@ -94,3 +94,31 @@ func deviceModelToUpdateDTO(device models.Device) dtos.UpdateDevice {
 		Notify:         &device.Notify,
 	}
 }
+
+// updateDeviceServiceCallback invoke device service's callback function for updating device service
+func updateDeviceServiceCallback(ctx context.Context, dic *di.Container, ds models.DeviceService) {
+	lc := container.LoggingClientFrom(dic.Get)
+	deviceServiceCallbackClient, err := newDeviceServiceCallbackClient(ctx, dic, ds.Name)
+	if err != nil {
+		lc.Errorf("fail to new a device service callback client by serviceName %s, err: %v", ds.Name, err)
+	}
+	dto := deviceServiceModelToUpdateDTO(ds)
+	response, err := deviceServiceCallbackClient.UpdateDeviceServiceCallback(ctx, requests.UpdateDeviceServiceRequest{Service: dto})
+	if err != nil {
+		lc.Errorf("fail to invoke device service callback for updating device service %s, err: %v", ds.Name, err)
+	}
+	if response.StatusCode != http.StatusOK {
+		lc.Errorf("fail to invoke device service callback for updating device service %s, err: %s", ds.Name, response.Message)
+	}
+}
+
+func deviceServiceModelToUpdateDTO(ds models.DeviceService) dtos.UpdateDeviceService {
```

Comment:
Shouldn't his be in core-contracts with the others here:
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/dtos/deviceservice.go#L54

Reply:
Yes, it should. Opened PR to create a transform function https://github.com/edgexfoundry/go-mod-core-contracts/pull/466.

Reply:
Rebased and upgrade the go-mod-core-contracts lib